### PR TITLE
Add perl buildsystem

### PR DESCRIPTION
**Pull request by satmandu**
## Description
- Adds perl buildsystem. Note that the package name internally is `PERL` so as not to conflict with the `perl` package.
- Rebuilds perl packages that had files in the `/usr/local/lib/perl5/5.38.0` dir.

## Additional information

Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=perl_buildsystem crew update
```

### CODE REVIEW

File: bin/crew [hunk 1]
```diff
@@ -657,7 +657,7 @@ def update
   # update package lists
   Dir.chdir(CREW_LIB_PATH) do
     # Set sparse-checkout folders.
-    system "git sparse-checkout set packages manifest/#{USER_SPACE_ARCH} lib bin crew tools"
+    system "git sparse-checkout set packages manifest/#{USER_SPACE_ARCH} lib bin crew tests tools"
```
Comment (Zopolis4) [inline bin/crew:660]: I've already got this change in #8993, best not to double up.

**Review by Zopolis4 — DISMISSED** [review]

File: lib/buildsystems/perl.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require 'package'
+
+class PErl < Package
```
Comment (uberhacker) [inline lib/buildsystems/perl.rb:3]: How about `PERL` all caps?  This mixed case is a bit confusing.

Comment (satmandu) [reply lib/buildsystems/perl.rb:3]: Sure.

File: lib/buildsystems/perl.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  property :pre_perl_options
+
+  def self.prebuild
+    puts "Additional pre_perl_options being used: #{@pre_perl_options.nil? ? '<no pre_perl_options>' : @pre_perl_options}".orange
+    system "#{@pre_perl_options} perl Makefile.PL"
+    system "sed -i 's,/usr/local,#{CREW_PREFIX},g' Makefile"
```
Comment (uberhacker) [inline lib/buildsystems/perl.rb:9]: Could this be in `def self.patch` or does that run before prebuild?

File: packages/perl_data_dumper.rb [hunk 4]
```diff
@@ -3,23 +3,23 @@
 class Perl_data_dumper < Package
```
Comment (uberhacker) [inline packages/perl_data_dumper.rb:3]: No buildsystems/perl?

File: packages/perl_file_spec.rb [hunk 5]
```diff
@@ -3,23 +3,23 @@
 class Perl_file_spec < Package
```
Comment (uberhacker) [inline packages/perl_file_spec.rb:3]: No buildsystems/perl?

File: packages/perl_data_dumper.rb (hunk 4 above)
Comment (satmandu) [reply packages/perl_data_dumper.rb:3]: The more complicated packages I did not change.

File: packages/perl_file_spec.rb (hunk 5 above)
Comment (satmandu) [reply packages/perl_file_spec.rb:3]: The more complicated packages I did not change.

File: lib/buildsystems/perl.rb (hunk 3 above)
Comment (satmandu) [reply lib/buildsystems/perl.rb:9]: patch runs before prebuild

File: packages/perl_data_dumper.rb (hunk 4 above)
Comment (satmandu) [reply packages/perl_data_dumper.rb]: changed

File: packages/perl_file_spec.rb (hunk 5 above)
Comment (satmandu) [reply packages/perl_file_spec.rb]: changed

**Review by uberhacker — APPROVED** [review]

**Pull request merged**